external-crates/move/crates/move-binary-format/src/compatibility.rs

```diff
@@ -360,98 +361,142 @@ pub enum InclusionCheck {
-        for (name, old_enum) in &old_module.enums {
-            let Some(new_enum) = new_module.enums.get(name) else {
-                return err;
-            };
-
-            if old_enum.abilities != new_enum.abilities {
-                return err;
-            }
-            if old_enum.type_parameters != new_enum.type_parameters {
-                return err;
-            }
-            if old_enum.variants.len() > new_enum.variants.len() {
-                return err;
+        // Since the structs are sorted we can iterate through each list to find the differences using two pointers
+        for mark in compare_ord_iters(old_module.structs.iter(), new_module.structs.iter()) {
+            match mark {
+                Mark::New(name, new) => context.struct_new(name, new),
+                Mark::Missing(name, _) => context.struct_missing(name),
+                Mark::Existing(name, old, new) => {
+                    if old != new {
+                        context.struct_change(name, old, new);
+                    }
+                }
             }
+        }
 
-            // NB: In the future if we allow adding new variants to enums in subset mode
-            // remove this if statement. This check is somewhat redundant with the one
-            // below, the one below should be kept if we allow adding variants in subset
-            // mode.
-            if old_enum.variants.len() != new_enum.variants.len() {
-                return err;
+        // enum checks
+        for mark in compare_ord_iters(old_module.enums.iter(), new_module.enums.iter()) {
+            match mark {
+                Mark::New(name, _) => context.enum_new(name, new_module.enums.get(name).unwrap()),
+                Mark::Missing(name, _) => context.enum_missing(name),
+                Mark::Existing(name, old, new) => {
+                    if old != new {
+                        context.enum_change(name, old);
+                    }
+                }
             }
+        }
 
-            if self == &Self::Equal && old_enum.variants.len() != new_enum.variants.len() {
-                return err;
-            }
-            // NB: We are using the fact that the variants are sorted by tag, that we've
-            // already ensured that the old variants are >= new variants, and the fact
-            // that zip will truncate the second iterator if there are extra there to allow
-            // adding new variants to enums in `Self::Subset` compatibility mode.
-            if !old_enum
-                .variants
-                .iter()
-                .zip(&new_enum.variants)
-                .all(|(old, new)| old == new)
-            {
-                return err;
+        // function checks
+        for mark in compare_ord_iters(old_module.functions.iter(), new_module.functions.iter()) {
+            match mark {
+                Mark::New(name, _) => {
+                    context.function_new(name, new_module.functions.get(name).unwrap())
```

Comment:
This can/should be updated to use the values returned from the `compare_ord_iters`